crates/swc_ecma_parser/src/parser/typescript.rs

```diff
@@ -2046,16 +2046,30 @@ impl<I: Tokens> Parser<I> {
             }
             tok!('-') => {
                 let start = cur_pos!(self);
+
                 bump!(self);
-                if !matches!(*cur!(self, true)?, Token::Num(..)) {
+
+                if !matches!(*cur!(self, true)?, Token::Num { .. }) {
                     unexpected!(self, "a numeric literal")
                 }
+
                 let lit = self.parse_lit()?;
                 let lit = match lit {
-                    Lit::Num(num) => TsLit::Number(Number {
-                        span: num.span,
-                        value: -num.value,
-                    }),
+                    Lit::Num(Number { span, value, raw }) => {
+                        let mut new_raw = String::from("-");
+                        let orig_raw = match raw {
+                            Some(raw) => raw,
+                            _ => value.to_string().into(),
```

Comment:
This line should be a call to `write!` macro.

Reply:
Current code allocates, and acquire global lock in `string_cache`, but we don't need these operations.